tests/testsuite/cargo_command.rs

```diff
@@ -380,3 +386,14 @@ fn closed_output_ok() {
     assert!(status.success());
     assert!(s.is_empty(), "{}", s);
 }
+
+#[cargo_test]
+fn subcommand_leading_plus_output_contains() {
+    cargo_process("+nightly")
+        .with_status(101)
+        .with_stderr_contains(
+            "\
+<tab>Cargo does not handle `+toolchain` directives.",
```

Comment:
Could you use `with_stderr` instead so we see the full message, acting more like a UI test, so we can ensure the message is formatted in an understandable way?

Reply:
Similarly, while most "did you mean" tests focus on different forms of detection, I'd like to see a "did you mean" test that uses `with_stderr` so we can have one verifying the UI

Reply:
> Similarly, while most "did you mean" tests focus on different forms of detection, I'd like to see a "did you mean" test that uses `with_stderr` so we can have one verifying the UI

A "did you mean" with the new message, correct?

Reply:
> A "did you mean" with the new message, correct?

Yes.  Currently, we only do `with_stderr_contains` for "did you mean" so we don't have UI coverage for the new message.  Most of those tests are probably focused the actual suggestion and probably don't make sense to turn into UI tests. Maybe an existing one does fit the bill, otherwise create a new test